File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-mktenv-5-4/CreditCurveValuation.cpp
// CreditCurveValuation.cpp 
#include "CreditCurveValuation.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

CreditCurveValuation::CreditCurveValuation(TiXmlNode* xmlNode)
: PricingStructureValuation(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //inputsNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* inputsNode = xmlNode->FirstChildElement("inputs");

   if(inputsNode){inputsIsNull_ = false;}
   else{inputsIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- inputsNode , address : " << inputsNode << std::endl;
   #endif
   if(inputsNode)
   {
      if(inputsNode->Attribute("href") || inputsNode->Attribute("id"))
      {
          if(inputsNode->Attribute("id"))
          {
             inputsIDRef_ = inputsNode->Attribute("id");
             inputs_ = boost::shared_ptr<QuotedAssetSet>(new QuotedAssetSet(inputsNode));
             inputs_->setID(inputsIDRef_);
             IDManager::instance().setID(inputsIDRef_,inputs_);
          }
          else if(inputsNode->Attribute("href")) { inputsIDRef_ = inputsNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { inputs_ = boost::shared_ptr<QuotedAssetSet>(new QuotedAssetSet(inputsNode));}
   }

   //defaultProbabilityCurveNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* defaultProbabilityCurveNode = xmlNode->FirstChildElement("defaultProbabilityCurve");

   if(defaultProbabilityCurveNode){defaultProbabilityCurveIsNull_ = false;}
   else{defaultProbabilityCurveIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- defaultProbabilityCurveNode , address : " << defaultProbabilityCurveNode << std::endl;
   #endif
   if(defaultProbabilityCurveNode)
   {
      if(defaultProbabilityCurveNode->Attribute("href") || defaultProbabilityCurveNode->Attribute("id"))
      {
          if(defaultProbabilityCurveNode->Attribute("id"))
          {
             defaultProbabilityCurveIDRef_ = defaultProbabilityCurveNode->Attribute("id");
             defaultProbabilityCurve_ = boost::shared_ptr<DefaultProbabilityCurve>(new DefaultProbabilityCurve(defaultProbabilityCurveNode));
             defaultProbabilityCurve_->setID(defaultProbabilityCurveIDRef_);
             IDManager::instance().setID(defaultProbabilityCurveIDRef_,defaultProbabilityCurve_);
          }
          else if(defaultProbabilityCurveNode->Attribute("href")) { defaultProbabilityCurveIDRef_ = defaultProbabilityCurveNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { defaultProbabilityCurve_ = boost::shared_ptr<DefaultProbabilityCurve>(new DefaultProbabilityCurve(defaultProbabilityCurveNode));}
   }

   //recoveryRateNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* recoveryRateNode = xmlNode->FirstChildElement("recoveryRate");

   if(recoveryRateNode){recoveryRateIsNull_ = false;}
   else{recoveryRateIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- recoveryRateNode , address : " << recoveryRateNode << std::endl;
   #endif
   if(recoveryRateNode)
   {
      if(recoveryRateNode->Attribute("href") || recoveryRateNode->Attribute("id"))
      {
          if(recoveryRateNode->Attribute("id"))
          {
             recoveryRateIDRef_ = recoveryRateNode->Attribute("id");
             recoveryRate_ = boost::shared_ptr<XsdTypeDecimal>(new XsdTypeDecimal(recoveryRateNode));
             recoveryRate_->setID(recoveryRateIDRef_);
             IDManager::instance().setID(recoveryRateIDRef_,recoveryRate_);
          }
          else if(recoveryRateNode->Attribute("href")) { recoveryRateIDRef_ = recoveryRateNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { recoveryRate_ = boost::shared_ptr<XsdTypeDecimal>(new XsdTypeDecimal(recoveryRateNode));}
   }

   //recoveryRateCurveNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* recoveryRateCurveNode = xmlNode->FirstChildElement("recoveryRateCurve");

   if(recoveryRateCurveNode){recoveryRateCurveIsNull_ = false;}
   else{recoveryRateCurveIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- recoveryRateCurveNode , address : " << recoveryRateCurveNode << std::endl;
   #endif
   if(recoveryRateCurveNode)
   {
      if(recoveryRateCurveNode->Attribute("href") || recoveryRateCurveNode->Attribute("id"))
      {
          if(recoveryRateCurveNode->Attribute("id"))
          {
             recoveryRateCurveIDRef_ = recoveryRateCurveNode->Attribute("id");
             recoveryRateCurve_ = boost::shared_ptr<TermCurve>(new TermCurve(recoveryRateCurveNode));
             recoveryRateCurve_->setID(recoveryRateCurveIDRef_);
             IDManager::instance().setID(recoveryRateCurveIDRef_,recoveryRateCurve_);
          }
          else if(recoveryRateCurveNode->Attribute("href")) { recoveryRateCurveIDRef_ = recoveryRateCurveNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { recoveryRateCurve_ = boost::shared_ptr<TermCurve>(new TermCurve(recoveryRateCurveNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<QuotedAssetSet> CreditCurveValuation::getInputs()
{
   if(!this->inputsIsNull_){
        if(inputsIDRef_ != ""){
             return boost::shared_static_cast<QuotedAssetSet>(IDManager::instance().getID(inputsIDRef_));
        }else{
             return this->inputs_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<QuotedAssetSet>();
   }
}
boost::shared_ptr<DefaultProbabilityCurve> CreditCurveValuation::getDefaultProbabilityCurve()
{
   if(!this->defaultProbabilityCurveIsNull_){
        if(defaultProbabilityCurveIDRef_ != ""){
             return boost::shared_static_cast<DefaultProbabilityCurve>(IDManager::instance().getID(defaultProbabilityCurveIDRef_));
        }else{
             return this->defaultProbabilityCurve_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<DefaultProbabilityCurve>();
   }
}
boost::shared_ptr<XsdTypeDecimal> CreditCurveValuation::getRecoveryRate()
{
   if(!this->recoveryRateIsNull_){
        if(recoveryRateIDRef_ != ""){
             return boost::shared_static_cast<XsdTypeDecimal>(IDManager::instance().getID(recoveryRateIDRef_));
        }else{
             return this->recoveryRate_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeDecimal>();
   }
}
boost::shared_ptr<TermCurve> CreditCurveValuation::getRecoveryRateCurve()
{
   if(!this->recoveryRateCurveIsNull_){
        if(recoveryRateCurveIDRef_ != ""){
             return boost::shared_static_cast<TermCurve>(IDManager::instance().getID(recoveryRateCurveIDRef_));
        }else{
             return this->recoveryRateCurve_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<TermCurve>();
   }
}
}

File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-mktenv-5-4/CreditCurveValuation.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class CreditCurveValuation : PricingStructureValuation
    {
        public CreditCurveValuation(XmlNode xmlNode)
        : base(xmlNode)
        {
            XmlNode inputsNode = xmlNode.SelectSingleNode("inputs");
            
            if (inputsNode != null)
            {
                if (inputsNode.Attributes["href"] != null || inputsNode.Attributes["id"] != null) 
                {
                    if (inputsNode.Attributes["id"] != null) 
                    {
                        inputsIDRef_ = inputsNode.Attributes["id"].Value;
                        QuotedAssetSet ob = new QuotedAssetSet(inputsNode);
                        IDManager.SetID(inputsIDRef_, ob);
                    }
                    else if (inputsNode.Attributes["href"] != null)
                    {
                        inputsIDRef_ = inputsNode.Attributes["href"].Value;
                    }
                    else
                    {
                        inputs_ = new QuotedAssetSet(inputsNode);
                    }
                }
                else
                {
                    inputs_ = new QuotedAssetSet(inputsNode);
                }
            }
            
        
            XmlNode defaultProbabilityCurveNode = xmlNode.SelectSingleNode("defaultProbabilityCurve");
            
            if (defaultProbabilityCurveNode != null)
            {
                if (defaultProbabilityCurveNode.Attributes["href"] != null || defaultProbabilityCurveNode.Attributes["id"] != null) 
                {
                    if (defaultProbabilityCurveNode.Attributes["id"] != null) 
                    {
                        defaultProbabilityCurveIDRef_ = defaultProbabilityCurveNode.Attributes["id"].Value;
                        DefaultProbabilityCurve ob = new DefaultProbabilityCurve(defaultProbabilityCurveNode);
                        IDManager.SetID(defaultProbabilityCurveIDRef_, ob);
                    }
                    else if (defaultProbabilityCurveNode.Attributes["href"] != null)
                    {
                        defaultProbabilityCurveIDRef_ = defaultProbabilityCurveNode.Attributes["href"].Value;
                    }
                    else
                    {
                        defaultProbabilityCurve_ = new DefaultProbabilityCurve(defaultProbabilityCurveNode);
                    }
                }
                else
                {
                    defaultProbabilityCurve_ = new DefaultProbabilityCurve(defaultProbabilityCurveNode);
                }
            }
            
        
            XmlNode recoveryRateNode = xmlNode.SelectSingleNode("recoveryRate");
            
            if (recoveryRateNode != null)
            {
                if (recoveryRateNode.Attributes["href"] != null || recoveryRateNode.Attributes["id"] != null) 
                {
                    if (recoveryRateNode.Attributes["id"] != null) 
                    {
                        recoveryRateIDRef_ = recoveryRateNode.Attributes["id"].Value;
                        XsdTypeDecimal ob = new XsdTypeDecimal(recoveryRateNode);
                        IDManager.SetID(recoveryRateIDRef_, ob);
                    }
                    else if (recoveryRateNode.Attributes["href"] != null)
                    {
                        recoveryRateIDRef_ = recoveryRateNode.Attributes["href"].Value;
                    }
                    else
                    {
                        recoveryRate_ = new XsdTypeDecimal(recoveryRateNode);
                    }
                }
                else
                {
                    recoveryRate_ = new XsdTypeDecimal(recoveryRateNode);
                }
            }
            
        
            XmlNode recoveryRateCurveNode = xmlNode.SelectSingleNode("recoveryRateCurve");
            
            if (recoveryRateCurveNode != null)
            {
                if (recoveryRateCurveNode.Attributes["href"] != null || recoveryRateCurveNode.Attributes["id"] != null) 
                {
                    if (recoveryRateCurveNode.Attributes["id"] != null) 
                    {
                        recoveryRateCurveIDRef_ = recoveryRateCurveNode.Attributes["id"].Value;
                        TermCurve ob = new TermCurve(recoveryRateCurveNode);
                        IDManager.SetID(recoveryRateCurveIDRef_, ob);
                    }
                    else if (recoveryRateCurveNode.Attributes["href"] != null)
                    {
                        recoveryRateCurveIDRef_ = recoveryRateCurveNode.Attributes["href"].Value;
                    }
                    else
                    {
                        recoveryRateCurve_ = new TermCurve(recoveryRateCurveNode);
                    }
                }
                else
                {
                    recoveryRateCurve_ = new TermCurve(recoveryRateCurveNode);
                }
            }
            
        
        }
        
    
        #region Inputs_
        private QuotedAssetSet inputs_;
        public QuotedAssetSet Inputs_
        {
            get
            {
                if (this.inputs_ != null)
                {
                    return this.inputs_; 
                }
                else if (this.inputsIDRef_ != null)
                {
                    inputs_ = IDManager.getID(inputsIDRef_) as QuotedAssetSet;
                    return this.inputs_; 
                }
                else
                {
                      return this.inputs_; 
                }
            }
            set
            {
                if (this.inputs_ != value)
                {
                    this.inputs_ = value;
                }
            }
        }
        #endregion
        
        public string inputsIDRef_ { get; set; }
        #region DefaultProbabilityCurve_
        private DefaultProbabilityCurve defaultProbabilityCurve_;
        public DefaultProbabilityCurve DefaultProbabilityCurve_
        {
            get
            {
                if (this.defaultProbabilityCurve_ != null)
                {
                    return this.defaultProbabilityCurve_; 
                }
                else if (this.defaultProbabilityCurveIDRef_ != null)
                {
                    defaultProbabilityCurve_ = IDManager.getID(defaultProbabilityCurveIDRef_) as DefaultProbabilityCurve;
                    return this.defaultProbabilityCurve_; 
                }
                else
                {
                      return this.defaultProbabilityCurve_; 
                }
            }
            set
            {
                if (this.defaultProbabilityCurve_ != value)
                {
                    this.defaultProbabilityCurve_ = value;
                }
            }
        }
        #endregion
        
        public string defaultProbabilityCurveIDRef_ { get; set; }
        #region RecoveryRate_
        private XsdTypeDecimal recoveryRate_;
        public XsdTypeDecimal RecoveryRate_
        {
            get
            {
                if (this.recoveryRate_ != null)
                {
                    return this.recoveryRate_; 
                }
                else if (this.recoveryRateIDRef_ != null)
                {
                    recoveryRate_ = IDManager.getID(recoveryRateIDRef_) as XsdTypeDecimal;
                    return this.recoveryRate_; 
                }
                else
                {
                      return this.recoveryRate_; 
                }
            }
            set
            {
                if (this.recoveryRate_ != value)
                {
                    this.recoveryRate_ = value;
                }
            }
        }
        #endregion
        
        public string recoveryRateIDRef_ { get; set; }
        #region RecoveryRateCurve_
        private TermCurve recoveryRateCurve_;
        public TermCurve RecoveryRateCurve_
        {
            get
            {
                if (this.recoveryRateCurve_ != null)
                {
                    return this.recoveryRateCurve_; 
                }
                else if (this.recoveryRateCurveIDRef_ != null)
                {
                    recoveryRateCurve_ = IDManager.getID(recoveryRateCurveIDRef_) as TermCurve;
                    return this.recoveryRateCurve_; 
                }
                else
                {
                      return this.recoveryRateCurve_; 
                }
            }
            set
            {
                if (this.recoveryRateCurve_ != value)
                {
                    this.recoveryRateCurve_ = value;
                }
            }
        }
        #endregion
        
        public string recoveryRateCurveIDRef_ { get; set; }
        public string choiceStr_0;
        
    
        
    
    }
    
}

